src/layouts/Settings.jsx

```diff
@@ -88,45 +92,45 @@ const CONFIG_FIELDS = [
 const FOOTER_FIELDS = ["otherFooterSettings", "socialMediaContent"]
 const NAVIGATION_FIELDS = ["navigationSettings"]
 
-export default class Settings extends Component {
-  _isMounted = false
-
-  constructor(props) {
-    super(props)
-    this.state = {
-      colorPicker: {
-        colorPickerToggle: false,
-        currentColor: "",
-        elementId: "",
-        oldColors: {},
-      },
-      siteName: "",
-      ...stateFields,
-      errors: _.cloneDeep(stateFields),
-    }
-  }
-
-  async componentDidMount() {
-    this._isMounted = true
-    try {
-      const { match } = this.props
-      const { siteName } = match.params
-      if (this._isMounted)
-        this.setState((currState) => ({
-          ...currState,
-          siteName,
-        }))
-
-      // get settings data from backend
-      const resp = await axios.get(
-        `${process.env.REACT_APP_BACKEND_URL_V2}/sites/${siteName}/settings`,
-        {
-          withCredentials: true,
-        }
-      )
-      const { configSettings, footerSettings, navigationSettings } = resp.data
+const Settings = ({ match, location }) => {
+  const { params, decodedParams } = match
+  const { siteName } = decodedParams
+
+  const [currState, setCurrState] = useState(stateFields)
+  const [originalState, setOriginalState] = useState()
+  const [hasErrors, setHasErrors] = useState(false)
+  const [errors, setErrors] = useState(
+    _.omit(_.cloneDeep(stateFields), "colors")
+  )
+  const [hasChanges, setHasChanges] = useState(false)
+  const [hasSettingsChanged, setHasSettingsChanged] = useState(false)
+  const [settingsStateDiff, setSettingsStateDiff] = useState()
+  const [showOverwriteWarning, setShowOverwriteWarning] = useState()
+
+  const { data: settingsData } = useGetSettingsHook(params)
+
+  const {
+    mutateAsync: updateSettingsHandler,
+    isLoading: isSavingSettings,
+  } = useUpdateSettingsHook(params, {
+    onError: (err) => {
+      if (err.response.status === 409) setShowOverwriteWarning(true)
+    },
+    onSuccess: () => {
+      setHasChanges(false)
+      setHasSettingsChanged(false)
+    },
+  })
+
+  useEffect(() => {
+    if (settingsData && !hasChanges) {
+      const {
+        configSettings,
+        footerSettings,
+        navigationSettings,
+      } = settingsData
 
-      const originalState = {
+      const retrievedState = {
```

Comment:
nit - is there a problem with using the `settingsData`'s `configSettings`, `footerSettings` and `navigationSettings` as is, instead of unpacking the values individually into new objects `otherFooterSettings`, `socialMediaContent` etc? there's quite a lot of state renaming and unpacking going on here, which makes it hard to follow what is happening to each value.

Reply:
If necessary, it could be useful to coordinate the backend data structure with the frontend so that we don't have to unpack and rename all our values multiple times.

Reply:
Good suggestion - updated in https://github.com/isomerpages/isomercms-frontend/pull/666/commits/2c15260967dcd709236a71ade747192e47415212